herd/libdir/aarch64util.cat

```diff
@@ -108,3 +120,24 @@ let Imp = NExp
 let SPONTANEOUS = SPURIOUS
 let inv-scope = inv-domain
 let TTD = PTE
+let Tag = T
+
+(* Same-cache-line relation *)
+(* NOTE : currently assumes all locations are on different cache lines *)
+(*        also extends to IC without a location specified              *)
+let scl = loc
+  | (M | DC.CVAU | IC) * (IC.IALLU | IC.IALLUIS)
+  | (IC.IALLU | IC.IALLUIS) * (M | DC.CVAU | IC)
+
+(* Flag any occurrence of writes to two different locations that might be on
+ * the same cache line. *)
+flag ~empty ((W & Instr)*(W & Instr) \ loc)
+  as Assuming-no-two-modified-instructions-are-on-the-same-cache-line
+
+(* Flag the runs of herd7 with unsupported combinations of variants *)
+flag ~empty (if "vmsa" && "memtag" then _ else 0)
+  as combining-vmsa-and-memtag-is-not-supported
+flag ~empty (if "vmsa" && "ifetch" then _ else 0)
+  as combining-vmsa-and-ifetch-is-not-supported
+flag ~empty (if "memtag" && "ifetch" then _ else 0)
+  as combining-memtag-and-ifetch-is-not-supported
```

Comment:
Please add a new line at the end of the file.

Reply:
Done, thanks.